### PR TITLE
fix(mcdu): fix negative efob on flight plan page

### LINKED ISSUE #4688: EFOB , FULEPRED values mismatch
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-05-01T22:21:52+00:00",
  "ref": "master",
  "sha": "b63c33db9b402dd4188cdf341e224eeaa71d6a57",
  "actor": "MisterChocker",
  "event_name": "manual"
}
```

### Describe the bug

Screenshot (294)
Screenshot (295)
Screenshot (296)

EFOB and FUELPRED  page mismatch 

### Expected behavior

Same values on both pages 

### Steps to reproduce

while performing a couple of flights 

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

Parshva8888#1339

### LINKED ISSUE #6798: Negative EFOB and blank fuel pred window
### Aircraft Version

Development

### Build info

```json
{
  "bu{
  "built": "2022-02-24T11:13:11+00:00",
  "ref": "master",
  "sha": "27b2b71b4c9520b8343b690e0a24a825b4c609a0",
  "actor": "svengcz",
  "event_name": "manual",
  "pretty_release_name": "master:27b2b71b"
}
```

### Describe the bug

Unexpected (negative) EFOB value that is not in accord with simbrief.
Flight from Frankfurt to Gibraltar. EFOB -1.7
The fuel prediction window also turned blank.

### Expected behavior

Expected 5 tons left at arrival Gibraltar.

### Steps to reproduce

1. simbrief OFP from EDDF to LXGB with 3tons extra fuel.
2. OFP: EDDF ANEK1F ANEKI Y163 HERBI Y164 OLBEN N869 NEMOS DCT GVA DCT INCUS DCT BALSI UN852 DIVKO UN975 BCN UM985 PIMOS DCT LXGB
3. FL350, two engines operational
4. EFOB value of -1.7 and increases
5. press the soft key on the right side of the left MCDU in the fuel pred that shows the ZFW value.
6. The MCDU will now show a black screen on the FUEL PRED window

### References (optional)

image
image

### Additional info (optional)

_No response_

### Discord Username (optional)

Warpig1017 / Roland#0520

### LINKED ISSUE #7449: EFOB on MCDUs FPLAN page not updated
### Aircraft Version

Development

### Build info

```json
{
    "built": "2022-08-27T08:10:52+00:00",
    "ref": "refs/heads/master",
    "sha": "3800df26d6879ba756abfe7ff22f5c4a97d53968",
    "actor": "2hwk",
    "event_name": "manual",
    "pretty_release_name": "master:3800df26"
}
```

### Describe the bug

The EFOB on the MCDUs FPLAN page is not in sync with the EFOB on the FUEL PRED page.

### Expected behavior

EFOB on FPLAN page and FUEL PRED page should be the same.

### Steps to reproduce

1. Fly some distance
2. Open FPLAN page and note EFOB (picture 1)
3. Open PRED FUEL page and note EFOB. Mostly EFOB differs from that on FPLAN page (picture 2)
4. Switch back to FPLAN page. Now the EFOB is set to the same as on PRED FUEL page (picture 
Screenshot 2022-08-27 225323
Screenshot 2022-08-27 225345
Screenshot 2022-08-27 225402
3

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

schroew#0206

### PR DESCRIPTION
**Pull request by Revyn112**
Fixes #7449 
Fixes #6798 
Fixes #4688
## Summary of Changes
When the flight plan page is active in the mcdu the EFOB will not be updated. This is caused by only updating the `_routeTripFuelWeight` when the fuel pred page is active. Also the MLW will not being updated on the EFB with correct dest efob for Payload Calculations, With this change the `tryUpdateRouteTrip` will be added to the flight plan page aswell. 

## Additional context

Discord username (if different from GitHub):
Revyn112#1010

## Testing instructions

1. Activate DevTools
2. Open Tools -> Behaviors
3. Switch to LocalVariables and monitor "A32NX_DESTINATION_FUEL_ON_BOARD" 
4. Init and setup your flight like you always do
5. After FUEL PRED Page is setup and F-PLN / FUEL PRED page is active the "A32NX_DESTINATION_FUEL_ON_BOARD" should be updated as soon as there is fuel consumption (APU/ENGINE).
6. As soon as you're in flight and the F-PLN / FUEL PRED page is active the EFOB should match the "A32NX_DESTINATION_FUEL_ON_BOARD" value.
(The EFOB will not change when still on ground).

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js [hunk 1]
```diff
@@ -683,7 +683,7 @@ class CDUFlightPlanPage {
             let destTimeCell = "----";
             let destDistCell = "---";
             let destEFOBCell = "---";
-
+
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:686]:
```suggestion

```
Please remove the trailing whitespace.

**Review by tracernz — COMMENTED** [review]
👍

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js (hunk 1 above)
Comment (Revyn112) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js]: done

**Review by tracernz — APPROVED** [review]

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord : 2Cas#1022
Tier of Testing  : 1
Date: 19/03/23

Testing Process: 
- [x] monitor "A32NX_DESTINATION_FUEL_ON_BOARD"
- [x] Init and setup your flight like you always do
- [x] After FUEL PRED Page is setup and F-PLN / FUEL PRED page is active the "A32NX_DESTINATION_FUEL_ON_BOARD" should be updated as soon as there is fuel consumption (APU/ENGINE).
- [x] As soon as you're in flight and the F-PLN / FUEL PRED page is active the EFOB should match the "A32NX_DESTINATION_FUEL_ON_BOARD" value. (The EFOB will not change when still on ground).

Negatives:
 - A32NX_DESTINATION_FUEL_ON_BOARD variable is not updated. EFOB on page display is updated however. 
 - Note that this variable is being removed from the payload backend regardless by #7715 so this is a non-issue.

Testing Results: 
Passed

Conclusions: 
LGTM

**Pull request merged**